trail/upload/process_uploads/header_standardizer.py

```diff
@@ -0,0 +1,329 @@
+
+        Parameters
+        ----------
+        header : `object`
+             The header, Astropy HDU and its derivatives.
+        \**kwargs : `dict`
+            Additional keyword arguments
+
+        Keyword arguments
+        -----------------
+        filename : `str`
+            Name of the file from which the HDU was read from, sometimes can encode
+            additional metadata.
+
+        Returns
+        -------
+        standardizerCls : `cls`
+            Standardizer class that can process the given upload.`
+
+        Raises
+        ------
+        ValueError
+            None of the registered processors can process the  upload.
+        """
+        standardizers = []
+        for standardizer in cls.standardizers.values():
+            if standardizer.canStandardize(header):
+                standardizers.append(standardizer)
+        standardizers.sort(key=lambda standardizer: standardizer.priority, reverse=True)
+
+        if standardizers:
+            if len(standardizers) > 1:
+                # I think this should never be an issue really, but just in case
+                names = [proc.name for proc in standardizers]
+                warnings.warn("Multiple standardizers declared ability to process "
+                              f"the given upload: {names}. \n Using {names[-1]} "
+                              "to process FITS.")
+            return standardizers[0]
+        else:
+            raise ValueError("None of the known standardizers can handle this upload.\n "
+                             f"Known standardizers: {list(cls.standardizers.keys())}")
+
+    @classmethod
+    def fromHeader(cls, header, **kwargs):
+        """Get the standardizer instance from a given header.
+
+        Parameters
+        ----------
+        header : `object`
+             The header, Astropy HDU and its derivatives.
+        \**kwargs : `dict`
+            Additional keyword arguments
+
+        Keyword arguments
+        -----------------
+        filename : `str`
+            Name of the file from which the HDU was read from, sometimes can encode
+            additional metadata.
+
+        Returns
+        -------
+        standardizerCls : `cls`
+            Standardizer class that can process the given upload.`
+
+        Raises
+        ------
+        ValueError
+            None of the registered processors can process the  upload.
+        """
+        # TODO: get some error handling here
+        standardizerCls = cls.getStandardizer(header, **kwargs)
+        return standardizerCls(header, **kwargs)
+
+    @abstractmethod
+    def standardizeMetadata(self):
+        """Normalizes FITS header information of the primary header unit and
+        returns a dictionary with standardized, as understood by trailblazer,
+        keys.
+
+        Returns
+        -------
+        standardizedKeys : `dict`
+          A dictionary with standardized header keys and values.
+        """
+        raise NotImplemented()
```

Comment:
Clarify somewhere that it's not implemented because it's instrument-specific, please

Reply:
I assume you want this for all abstract base classes methods? Because they will complain quite explicitly when someone tries to define a class without defining all the required abstract methods, I think something like: `Can't instantiate abstract class with abstract methods ...` thing.

Reply:
Oh that's fair. I meant just in the docstring in these "standard" standardizers, saying explicitly they're ABC methods intended to be overridden (inherited? whatever the right word is) by an instrument-specific one instead. Maybe that's implied by how you wrote it and I'm just not as used to the convention as I should be.

Reply:
No problems with adding the comments, I just was not clear if you meant this one specific method or all abstract methods in the code?

Reply:
Ideally all of them. Because I'm bad at telling which ones are this flavor vs an actual functionality we haven't implemented yet 🙂 It can be the same generic one line comment.

Reply:
Ok, will add tonight. None of the methods that currently raise `NotImplementedError` are raising them because it's a thing that isn't actually implemented yet. For some, like MOA, they are fake and should be fixed, for others they actually work. Mostly they raise them because they are meant to be overridden once inherited from the class, at which point they inherit the docstring of the method too. Or so I hope (the docstring bit).